### PR TITLE
Jvenstad/configurable tester flavours

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review, but we should probably have a discussion before merging this.

### CODE REVIEW

**Comment by jonmv** [comment]
This makes the tester flavour configurable per environment, like the Athenz service. Perhaps unnecessarily fine-grained? Or should it be per region? I don't think so ...

File: config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java [hunk 1]
```diff
@@ -247,11 +266,15 @@ private boolean trueOrMissing(String value) {
 
         String policy = upgradeElement.getAttribute("policy");
         switch (policy) {
-            case "canary" : return DeploymentSpec.UpgradePolicy.canary;
-            case "default" : return DeploymentSpec.UpgradePolicy.defaultPolicy;
-            case "conservative" : return DeploymentSpec.UpgradePolicy.conservative;
-            default : throw new IllegalArgumentException("Illegal upgrade policy '" + policy + "': " +
-                                                         "Must be one of " + Arrays.toString(DeploymentSpec.UpgradePolicy.values()));
+            case "canary":
```
Comment (bratseth) [inline config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java]: This reformatting just makes it harder to read ...

**Review by bratseth — APPROVED** [review]
Looks fine to me. Before we make it public we should bring it up in an arch meeting.

File: config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java (hunk 1 above)
Comment (jonmv) [reply config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java]: Completely agree with you. I didn't notice this snuck in. IntelliJ loves doing this.

**Comment by jonmv** [comment]
The one thing to consider is whether we will have so many test-related attributes they should be grouped in a `<test>` element, or something. What you do think?

File: config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java (hunk 1 above)
Comment (bratseth) [reply config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java]: Please don't autoformat ...

**Comment by bratseth** [comment]
I don't know. The counter argument to doing it eagerly is that the structure becomes harder to understand with too many tags with different semantics mixed together. We can discuss it on Thursday.

File: config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java (hunk 1 above)
Comment (jonmv) [reply config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java]: Well, I never do on purpose. I don't even know what change may have caused IntelliJ to jump in here.

Comment (jonmv) [reply config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java]: Looks very much like auto-format was triggered somehow — doing it now does nothing 💡 
Any idea what may have caused it to trigger?

**Comment by jonmv** [comment]
Yes, I lean towards the simplest (attribute) approach. The only other settings I can think of right now are one or two timeouts. 

So, merge?

File: config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java (hunk 1 above)
Comment (bratseth) [reply config-model-api/src/main/java/com/yahoo/config/application/api/xml/DeploymentSpecXmlReader.java]: No. I've turned that off.

**Pull request merged**